src/Commands/TankDrive.cpp

```diff
@@ -42,6 +42,30 @@ void TankDrive::Execute() {
 	RobotMap::chassisrightMotor2->Set(rY);
 	RobotMap::chassisrightMotor3->Set(rY);
 
+	//Automatic Shifters
+	SmartDashboard::PutNumber("EncodeValue chassisrightmotor2",
+			RobotMap::chassisrightMotor2->GetEncVel());
+	double shiftUpVelocity = SmartDashboard::GetNumber("UpVelocity", 3000);
+
+	double shiftDownVelocity = SmartDashboard::GetNumber("DownVelocity", 1000);
+
+	if ((RobotMap::chassisshift->Get() == DoubleSolenoid::kReverse)
+			&& (RobotMap::chassisleftMotor1->GetEncVel()
+			|| RobotMap::chassisrightMotor2->GetEncVel() >= shiftUpVelocity)
+		 && (Robot::oi->getdriver()->GetRawAxis(1)
+				|| Robot::oi->getdriver()->GetRawAxis(5) >= .8) )
```

Comment:
`const double kShiftThreshold = 0.8; ` ? A name that makes it obvious what it affects.